src/FadingCounter.h

```diff
@@ -15,26 +15,28 @@
 class FadingCounter
 {
 public:
+    static const size_t Precision = 10; ///< #counting slots, controls measur. accuracy
```

Comment:
Your comment implies that this is something new. It is not `precision` has always existed and never modified after object initialization to a value of `10`. Making it a `static const` just reduces the needless memory consumption.

Reply:
> Your comment implies that this is something new. It is not `precision` has always existed and never modified after object initialization to a value of `10`.

That "something new" here is a design change from "each counter may have its own precision (that never changes)" design to "all counters have identical precision (that never changes)" design. The former design does not allow std::array usage. The new design does. The fact that the old design used a constant 10 to initialize that per-counter precision is not important (from the design point of view).

BTW, each design has pros/cons, of course. I am not attacking or defending a particular design in this change request. I am only saying that if this PR switches to a one-size-fits-all design, it should go much further.

> Making it a `static const` just reduces the needless memory consumption.

I disagree with the "just" part, and the next steps I outlined would remove a lot more waste than an integer data member! Also, such changes to private members seem to be outside this "update API" PR scope. This PR does not have to make this change, but if it does, it becomes responsible for seeing it through.

Reply:
> > "each counter may have its own precision (that never changes)"
> 
> There is/was no setter for the `private` member. So no, they could not have changed it from the constructed value of `10`. It has always been a constant but lacked the `const` marker.

The design aspect discussed here is not about _changing_ existing precision of a FadingCounter instance. It is about configuring custom precision at FadingCounter construction time. With the old design, it would be trivial to add a constructor (parameter) that initializes precision to a value other than 10 for a particular FadingCounter instance. The precision member would still be constant for that instance -- it will never change. With the new design, that customization becomes impossible because the precision value is shared across all FadingCounter instances.

> Your line of arguments are based on a design which several years worth of development have yet to identify a single case of need for the `precision` member to be non-const.

The above sounds as if I am defending the old design and/or trying to preserve it. I am not! My argument is not "the old design was good, so this PR should ...". Instead, my argument is "this PR _changes_ the old design, so this PR is responsible for the consequences of that change". Migrating from runtime-configurable std::vector to compiler-sized std::array is such a consequence.

Thank you for addressing my concern.